lib/events/user.ts

```diff
@@ -11,25 +13,35 @@ export default class UserEvent extends EventEmitter {
     this.crowi = crowi
   }
 
-  async onActivated(user) {
+  async onActivated(user: UserDocument) {
     const Page = this.crowi.model('Page')
-
     const userPagePath = Page.getUserPagePath(user)
-    Page.findPage(userPagePath, user, {}, false)
-      .then(function (page) {
-        // do nothing because user page is already exists.
-      })
-      .catch(function (err) {
-        const body = `# ${user.username}\nThis is ${user.username}'s page`
-        // create user page
-        Page.createPage(userPagePath, body, user, {})
-          .then(function (page) {
-            // page created
-            debug('User page created', page)
-          })
-          .catch(function (err) {
-            debug('Failed to create user page', err)
-          })
-      })
+
+    try {
+      const page = await Page.findPage(userPagePath, user, {}, false)
+
+      if (page === null) {
+        throw new Error()
+      }
+
+      // User page is already exists.
+      const renamedUserPagePath = `/tmp/user-${user.username}-${Date.now()}`
+      await Page.rename(page, renamedUserPagePath, user, {})
+      await this.createUserPage(userPagePath, user)
+    } catch {
+      await this.createUserPage(userPagePath, user)
```

Comment:
💬 If you handle page existence with throwing `Error`, we won't be able to handle it separately from other types of errors 🙁 (e.g. failing page rename).
It would be better if handle the page existence just with control flow.

Reply:
Good suggestion!
I noticed `ignoreNotFound` option that `Page.findPage` return `null` instead of throwing `Error`.